python/ambassador/envoy/v2/v2listener.py

```diff
@@ -1077,21 +1077,19 @@ def generate(cls, config: 'V2Config') -> None:
                         variant = "secure" if secure else "insecure"
 
                         if route["match"].get("prefix", None) == "/.well-known/acme-challenge/":
-                            # ACME challenges can never be secure.
+                            # We need to be sure to route ACME challenges, no matter what else is going
+                            # on (this is the infamous ACME hole-puncher mentioned everywhere).
+                            logger.debug(f"V2Listeners: {listener.name} {vhostname} force Route for ACME challenge")
+                            action = "Route"
+
+                            # We have to force the correct route entry, too, just in case. (Note that right now,
+                            # the user can't create a Mapping that forces redirection. When they can do this
+                            # per-Mapping, well, really, we can't force them to not redirect if they explicitly
+                            # ask for it, and that'll be OK.)
+
                             if secure:
-                                logger.debug(
-                                    f"V2Listeners: {listener.name} {vhostname} secure: force Reject for ACME challenge")
-                                action = "Reject"
+                                route = secure_route
```

Comment:
Is it even possible for `route` to not already be `secure_route` if we get here? (and for `action` to not already be `"Route"` a few lines up?

Am I correct in thinking that
```python
if route["match"].get("prefix", None) == "/.well-known/acme-challenge/":
```
could become
```
if route["match"].get("prefix", None) == "/.well-known/acme-challenge/" and not secure:
```
and then drop the `if secure` bit?  And that it wouldn't change behavior at all?

Reply:
Flynn says he is 99.5% certain that what I said is accurate.